pkg/urbit/vere/io/ames.c

```diff
@@ -435,6 +598,76 @@ _ames_hear_bail(u3_ovum* egg_u, u3_noun lud)
   u3_ovum_free(egg_u);
 }
 
+/* _ames_put_packet(): add packet to queue, drop old packets on pressure
+*/
+static void
+_ames_put_packet(u3_ames* sam_u,
+                 u3_noun  msg,
+                 u3_noun  lan)
+{
+  u3_noun wir = u3nc(c3__ames, u3_nul);
+  u3_noun cad = u3nt(c3__hear, u3nc(c3n, lan), msg);
+
+  u3_auto_peer(
+    u3_auto_plan(&sam_u->car_u,
+                 u3_ovum_init(0, c3__a, wir, cad)),
+    0, 0, _ames_hear_bail);
+
+  _ames_cap_queue(sam_u);
+}
+
+/*  _ames_forward(): forward pac_u onto the (list lane) las, then free pac_u
+*/
+static void
+_ames_forward(u3_panc* pac_u, u3_noun las)
+{
+  pac_u->sam_u->fow_d++;
+  if ( 0 == (pac_u->sam_u->fow_d % 10000) ) {
+    u3l_log("ames: forwarded %" PRIu64 " total\n", pac_u->sam_u->fow_d);
+  }
+
+  {
+    u3_noun pac = _ames_serialize_packet(pac_u, c3y);
+    while (u3_nul != las) {
+      _ames_ef_send(pac_u->sam_u, u3h(las), u3k(pac));
+      las = u3t(las);
+    }
+    u3z(pac);
+  }
+
+  pac_u->sam_u->foq_d--;
+  _ames_panc_free(pac_u);
+}
+
+/*  _ames_lane_scry_cb(): learn lane to forward packet on
+*/
+static void
+_ames_lane_scry_cb(void* vod_p, u3_noun nun)
+{
+  u3_panc* pac_u = vod_p;
+  u3_weak  las = u3r_at(15, nun);  //TODO  why [~ %noun ~ lane]
```

Comment:
@Fang-, this was just a refactoring bug: introduced in a53f688d01262403a5c61bb9a68ca243cf080411 and fixed in 406c86568900b7be0c2bea0239fe8806a7a0f010. Please merge `ipc-redux` back into this branch and correct the axis.